### PR TITLE
chore: remove unused code and add linting and pre-commit hooks

### LINKED ISSUE #502: clean up unused code + add linter for unused variables/imports/etc
there's a lot of leftover/unused code from removing antd. This should all be removed

Additionally, we should remove any unused variables, functions, imports, etc. and add linting for this

### PR DESCRIPTION
**Pull request by katiestahl**
close #502

### CODE REVIEW

**Comment by jsstevenson** [comment]
fwiw, I think some of the code from your earlier PRs is in here, but you can consider this as an "approval" for this PR once those other ones are finished and merged into dev

**Review by mcannon068nw — APPROVED** [review]
approved for once the other PRs are in

**Comment by katiestahl** [comment]
> fwiw, I think some of the code from your earlier PRs is in here, but you can consider this as an "approval" for this PR once those other ones are finished and merged into dev

thanks for catching this. I think I accidentally committed some code when I was playing around with breadcrumbs. I removed it for now.

**Pull request merged**
